### ISSUE TITLE
Low Inference accuracy after training alexnet_v2 using tf.slim - Misses a proper preprocessing script

### ISSUE THREAD

**Issue by amirjamez** [tensorflow/models#5311]
System information:
Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 14.04
TensorFlow installed from (source or binary): Source
TensorFlow version (use command below): r1.8
Python version: 2.7
Bazel version (if compiling from source): bazel release 0.13.0
GCC/Compiler version (if compiling from source): 4.8
CUDA/cuDNN version: 8/7
GPU model and memory: NVIDIA-1060

I have trained `alexnet_v2` available in `tf.slim` by manually making a preprocessing file based on #423 suggestion. However, after 200K step the accuracy is almost 0:

```
2018-09-15 13:15:04.595764: I tensorflow/core/kernels/logging_ops.cc:79] eval/Recall_4[0.004]
2018-09-15 13:15:04.595803: I tensorflow/core/kernels/logging_ops.cc:79] eval/Accuracy[0.001]
2018-09-15 13:15:04.596105: I tensorflow/core/kernels/logging_ops.cc:79] eval/Recall_5[0.005]
2018-09-15 13:15:04.596198: I tensorflow/core/kernels/logging_ops.cc:79] eval/Recall_3[0.003]
```

Here is the preprocessing file I manually put in `preprocessing/alexnet_preprocessing.py`:

```
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import tensorflow as tf

_PADDING = 4
slim = tf.contrib.slim

def preprocess_image(image, output_height, output_width, is_training=False):
  resized_image = tf.image.resize_image_with_crop_or_pad(image, output_width, output_height)
  tf.summary.image('image', tf.expand_dims(image, 0))
  return tf.image.per_image_standardization(resized_image)
```

Does it need some missing optimization functions? Thanks! @snnn @yifeif  @abes975 @sguada @jmchen-g

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by amirjamez** [comment]
Updated @tensorflowbutler

**Comment by k-w-w** [comment]
What command are you running?

**Comment by amirjamez** [comment]
@k-w-w I execute in the same `tf.slim` way:

```
python train_image_classifier.py     --train_dir=${TRAIN_DIR}     --dataset_name=imagenet     --dataset_split_name=train     --dataset_dir=${DATASET_DIR}     --model_name=alexnet_v2
```

`preprocessing/alexnet_preprocessing.py` has a dictionary of all preprocessing files, so I created one for `alexnet_v2` as mentioned in my first post. I could replicate and used what inception preprocessing does as the dimensions match, but I found at the learning take too long, so I was wondering if you guys have built an optimized preprocessing (possibly plus other learning parameters) so it trains faster.

**Comment by k-w-w** [comment]
Oh, I see, thanks for the clarification. Assigning @sguada his response the question.
Also adding the feature request tag since there appears to be multiple users interested in an alexnet preprocessing script.

**Comment by wx1111** [comment]
i meet the similar problem that the my accuracy is 0.38 and no longer change after 20 epochs , while the training accuracy is  0.78 after 100 epochs. my preprocessing script is just decode the tf-record files and then resize image.
my  preprocessing script:
    image = tf.image.decode_jpeg(image_buffer, channels=3)
    image = tf.image.resize_images(image,[224,224])

@k-w-w @sguada

**Comment by amirjamez** [comment]
@k-w-w @sguada @wx1111 I have ended up using inception preprocessing for that matter. After 10M steps it converged to 56% Top-1 and 79.8% Top-5. Accuracy didn't change much after 5M steps.

**Comment by wx1111** [comment]
@amirjamez  do you change any default parameters like the init learning rate or the learning rate decay policy when training the alexnet_v2 model from scartch on imagenet. i used the default parameters and the loss is very hard to converge. then i changed the batch size from 32 to 512 and the init learning rate from 0.01 to 0.1. it seems do not work well either.

**Comment by amirjamez** [comment]
@wx1111 I used the `--preprocessing_name=inception_preprocessing` and `--batch_size=32` and after one million iterations `Top-1` reached around 45-48% and `Top-5` around 70%. I let it train up to 5 million and I obtained more or less the same Top-1 and Top-5 reported in the paper. Hope this helps.

**Comment by rfgil** [comment]
@amirjamez Could you upload the checkpoint file you obtained? I would like to run some inference tests without spending much time training the network.

**Comment by amirjamez** [comment]
@rfgil sure. I have made a small repository with the training explanation and the pretrained model here as [https://github.com/amirjamez/alexnetTF]. Hope it helps :)

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
